docs/content/querying/dimensionspecs.md

```diff
@@ -371,3 +371,29 @@ Following filtered dimension spec retains only the values matching regex. Note t
 ```
 
 For more details and examples, see [multi-valued dimensions](multi-valued-dimensions.html).
+
+### Lookup DimensionSpecs
+
+Lookup DimensionSpecs can be used to define directly a lookup implementation as dimension spec.
+Generally speaking there is two different kind of lookups implementations. 
+The first kind is passed at the query time like `map` implementation.
+
+```json
+{ 
+  "type":"lookup"
+  "dimension":"dimensionName"
+  "outputName":"dimensionOutputName"
+  "lookup":{"type": "map", "map":{"key":"value"}, "isOneToOne":false}
```

Comment:
@drcrallen it is called `ONE_TO_ONE` https://github.com/druid-io/druid/blob/3459a202ce751cb60884519ef902e35280550895/processing/src/main/java/io/druid/query/extraction/ExtractionFn.java#L98-L98